generators/app/templates/src/app/components/CustomTextInput/index.js

```diff
@@ -0,0 +1,104 @@
+import styles from './styles';
+
+// Known issue: showing/hiding the text with secureTextEntry changes the cursor position
+// https://github.com/facebook/react-native/issues/5859
+
+class CustomTextInput extends React.Component {
+  state = { showPassword: false };
+
+  handleShowPassword = () => {
+    this.setState(prevState => ({ showPassword: !prevState.showPassword }));
+  };
+
+  render() {
+    const placeholderTextColor = this.props.value ? transparent : this.props.placeholderTextColor;
+
+    return (
+      <View>
+        {this.props.title && (
+          <CustomText darkGray style={[styles.title, this.props.titleStyles]}>
+            {this.props.title}
+          </CustomText>
+        )}
+        <View
+          style={[
+            this.props.multiline ? styles.multilineContainer : styles.container,
+            this.props.bottomBorder ? styles.bottomBorder : {},
+            this.props.style
+          ]}
+        >
+          <TextInput
+            {...this.props}
+            allowFontScaling={false}
+            onChangeText={this.props.onChange}
+            onBlur={this.props.onBlur}
+            onFocus={this.props.onFocus}
+            value={this.props.value}
+            style={[
+              styles.inputStyle,
+              this.props.multiline ? styles.multilineInput : styles.singleInput,
+              this.props.textStyles
+            ]}
+            placeholderTextColor={placeholderTextColor}
+            selectionColor={blue}
+            secureTextEntry={this.props.secureTextEntry && !this.state.showPassword}
+            autoComplete={!this.props.secureTextEntry}
+          />
+          {this.props.secureTextEntry &&
+            this.props.showEye && (
+              <ShowPassword
+                onShowPassword={this.handleShowPassword}
+                passwordVisible={this.state.showPassword}
+              />
+            )}
+        </View>
+      </View>
+    );
+  }
+}
+
+CustomTextInput.defaultProps = {
+  placeholder: this.emptyString,
+  autoCorrect: false,
+  autoCapitalize: 'sentences',
+  clearButtonMode: 'never',
+  returnKeyType: 'done',
+  underlineColorAndroid: transparent,
+  keyboardType: 'default',
+  multiline: false,
+  bottomBorder: false,
+  maxHeight: 200
+};
+
+CustomTextInput.propTypes = {
+  bottomBorder: PropTypes.bool,
+  autoCapitalize: PropTypes.oneOf('none', 'sentences', 'words', 'characters'),
+  clearButtonMode: PropTypes.oneOf('never', 'while-editing', 'unless-editing', 'always'),
+  returnKeyType: PropTypes.oneOf('done', 'go', 'next', 'search'),
+  keyboardType: PropTypes.oneOf('default', 'numeric', 'email-address', 'phone-pad'),
+  underlineColorAndroid: PropTypes.string,
+  maxHeight: PropTypes.number,
+  autoCorrect: PropTypes.bool,
+  multiline: PropTypes.bool,
+  placeholder: PropTypes.string,
+  titleStyles: Text.propTypes.style, // eslint-disable-line react/no-typos
+  textStyles: Text.propTypes.style, // eslint-disable-line react/no-typos
```

Comment:
Same comment here.
Mmm isn't this deprecated ? 🤔  I've searched for `TextPropTypes` but I've found nothing.
For example, `View.propTypes.style` is deprecated.

Reply:
View.propTypes.style is deprecated, but there's no info about Text.propTypes. and it doesnt throw a warning or sth